R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,717 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise**: Go to trainer-1's idr0021 project, copy the ID and paste it below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "projectId <- REPLACE_WITH_PROJECT_ID \n",
+    "project <- loadObject(server, \"ProjectData\", projectId)\n",
+    "dataframes <- availableDataframes(project)\n",
+    "print(dataframes)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load the dataframe\n",
+    "dfID <- dataframes$ID[1]\n",
+    "centrioles <- loadDataframe(project, dfID)\n",
+    "\n",
+    "# Make sure the data types are correct\n",
+    "centrioles$Dataset <- as.factor(centrioles$Dataset)\n",
+    "centrioles$Diameter <- as.numeric(centrioles$Diameter)\n",
+    "centrioles"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Plot the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Order the datasets ascending by mean diameter \n",
+    "ag <-aggregate(centrioles$Diameter ~ centrioles$Dataset, centrioles, median)\n",
+    "ordered <- factor(centrioles$Dataset, levels=ag[order(ag$`centrioles$Diameter`), 'centrioles$Dataset'])\n",
+    "\n",
+    "# Draw the plot \n",
+    "plot(centrioles$Diameter ~ ordered, ylab='Diameter (nm)', xlab=\"Protein\", cex.axis=0.5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Interpretation\n",
+    "\n",
+    "#### One-way analysis of variance\n",
+    "\n",
+    "Is there a significant difference between the proteins?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fit <- aov(centrioles$Diameter ~ centrioles$Dataset)\n",
+    "summary(fit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Two-sample Wilcoxon test of all pairwise combinations\n",
+    "\n",
+    "Are all proteins significantly different from each other?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Two-sample Wilcoxon test ('Mann-Whitney') of all pairwise combinations:\n",
+    "combins <- combn(levels(centrioles$Dataset), 2)\n",
+    "params_list <- split(as.vector(combins), rep(1:ncol(combins), each = nrow(combins)))\n",
+    "testResults <- data.frame()\n",
+    "for (param in params_list) {\n",
+    "  testdf <- subset(centrioles, centrioles$Dataset %in% param)\n",
+    "  pval <- wilcox.test(formula = Diameter ~ Dataset, data = testdf)$p.value\n",
+    "  testResults <- rbind(testResults, data.frame(Protein_1=param[1], Protein_2=param[2], p_value=pval))\n",
+    "}\n",
+    "testResults"
+   ]
```

Comment:
After running this cell, I get

```
Warning message in wilcox.test.default(x = c(211.522222470665, 214.349311216569, :
“cannot compute exact p-value with ties”
```

Yes, not too worrying, but of course people will ask what that is ?

Reply:
Yes, coincidentaly there are really two ROIs which are exactly(!) the same size and diameter, amazing.